libs/subliminal_patch/providers/whisperai.py

```diff
@@ -286,10 +313,26 @@ def download_subtitle(self, subtitle: WhisperAISubtitle):
         # TODO: This loads the entire file into memory, find a good way to stream the file in chunks
 
         out = encode_audio_stream(subtitle.video.original_path, self.ffmpeg_path, subtitle.force_audio_stream)
+        if out == None:
+            logger.info(f"Whisper cannot process {subtitle.video.original_path} because of missing/bad audio track")
+            subtitle.content = None
+            return
+
+        if subtitle.task == "transcribe":
+            output_language = subtitle.audio_language
+        else:
+            output_language = "eng"
+
+        logger.info(f'Starting WhisperAI {subtitle.task} to {language_from_alpha3(output_language)} for {subtitle.video.original_path}')
+        startTime = time.time()
 
         r = self.session.post(f"{self.endpoint}/asr",
-                              params={'task': subtitle.task, 'language': whisper_get_language_reverse(subtitle.audio_language), 'output': 'srt', 'encode': 'false'},
+                              params={'task': subtitle.task, 'language': whisper_get_language_reverse(output_language), 'output': 'srt', 'encode': 'false'},
```

Comment:
The language param refers to input language, not output.

If you translate and set "language" to english on a non-english audio file, what happens? I suspect it could fail.

Reply:
Good catch. Fixed.
In my experience, if you give whisper the wrong input language, it still does the translation into English properly, but it takes a bit longer and it might not be as quite as accurate. It still does work though.